lib/timers.js

```diff
@@ -554,56 +557,66 @@ function tryOnImmediate(immediate, queue) {
   }
 }
 
+function runCallback(timer) {
+  var argv = timer._argv;
+  var argc = argv ? argv.length : 0;
+  switch (argc) {
+    // fast-path callbacks with 0-3 arguments
+    case 0:
+      return timer._callback();
+    case 1:
+      return timer._callback(argv[0]);
+    case 2:
+      return timer._callback(argv[0], argv[1]);
+    case 3:
+      return timer._callback(argv[0], argv[1], argv[2]);
+    // more then 3 arguments run slower with .apply
+    default:
+      return timer._callback.apply(timer, argv);
+  }
+}
 
-function Immediate() { }
-
-Immediate.prototype.domain = undefined;
-Immediate.prototype._onImmediate = undefined;
-Immediate.prototype._idleNext = undefined;
-Immediate.prototype._idlePrev = undefined;
 
+function createImmediate(callback) {
+  return {
+    _idleNext: null,
+    _idlePrev: null,
+    _callback: callback,
+    _argv: null,
```

Comment:
it was small, nanoseconds, but I didn't record it and can't reproduce it now.
I re-timed passing two parameters to the create function and compiled into node
the two-argument form is distinctly faster, so I'll change this (my test rig loads
the file under test with `require`, so can run the same timers.js source under
various engines.  I'd been comparing v0.10.42 vs v4.4.0 vs v5.10.1 vs v6.0.0)